test/unit/middlewares/rateLimit.test.ts

```diff
@@ -42,13 +42,15 @@ describe('Rate Limiting', () => {
     jest.clearAllMocks()
     // Reset config to default state
     CONFIG.rateLimit = true
+    // Mock console.error
+    jest.spyOn(console, 'error').mockImplementation(() => {})
   })
 
   it('should skip rate limiting when disabled in config', async () => {
     CONFIG.rateLimit = false
 
     const mockReq = {
-      socket: { remoteAddress: '127.0.0.1' },
+      ip: '127.0.0.1',
```

Comment:
why is this impacted w/ these changes?

Reply:
the unit testing fails as the flow expects ip in the request post changes as per this [comment](https://github.com/shardeum/json-rpc-server/pull/91#discussion_r1862487043)